osbs/api.py

```diff
@@ -541,6 +543,31 @@ def get_token(self):
         return self.os.get_oauth_token()
 
     @osbsapi
+    def login(self, token):
+        token_file = utils.get_instance_token_file_name(self.os_conf.conf_section)
+        token_file_dir = os.path.dirname(token_file)
+
+        if not os.path.exists(token_file_dir):
+            os.makedirs(token_file_dir)
+
+        # Shameless copy from http://stackoverflow.com/a/15015748/5998718
+        # For security, remove file with potentially elevated mode
+        if os.path.exists(token_file):
+            os.remove(token_file)
+
+        # Open file descriptor
+        umask_original = os.umask(0)
```

Comment:
I'm actually not so sure we need to reset the umask. If the user wants to set their umask to something so restrictive they can't write to their own files, we should probably honour that.